astarte-device-sdk/AstarteDeviceSDK.cpp

```diff
@@ -393,3 +393,33 @@ bool AstarteDeviceSDK::disconnectFromAstarte()
 
   return m_astarteTransport->disconnectFromBroker();
 }
+
+template <typename T>
+bool AstarteDeviceSDK::sendData(const QByteArray &interface, const QByteArray &path, const QList<T> &value, const QDateTime &timestamp, const QVariantHash &metadata)
+{
+    QList<QVariant> value2;
+    value2.reserve(value.length());
+
+    for (auto i: value) {
```

Comment:
we should never call a variable `i` if it is not an integer index.

Reply:
as far as I know it is advised to use `for (int i = 0; i < len; i++)` when possible for containers instead of C++ range for, at least for Qt5.